libsolidity/ast/ASTUtils.cpp

```diff
@@ -107,4 +107,17 @@ Type const* type(VariableDeclaration const& _variable)
 	return _variable.annotation().type;
 }
 
+u256 storageLayoutBaseForInheritanceHierarchy(ContractDefinition const& _topLevelContract)
+{
+	solAssert(!_topLevelContract.abstract());
+	solAssert(!_topLevelContract.isInterface());
+	solAssert(!_topLevelContract.isLibrary());
+	if (auto const* storageLayoutSpecifier = _topLevelContract.storageLayoutSpecifier())
+	{
+		solAssert(storageLayoutSpecifier->annotation().baseSlot.set());
+		return *storageLayoutSpecifier->annotation().baseSlot;
+	}
```

Comment:
You don't generally have to assert this on `SetOnce`. `*` already asserts this so you can make things simpler:
```suggestion
		return *storageLayoutSpecifier->annotation().baseSlot;
```